components/infobox/wikis/ageofempires/infobox_league_custom.lua

```diff
@@ -356,48 +352,46 @@ function CustomLeague:_getGameModes(args, makeLink)
 end
 
 function CustomLeague:_getMaps(args)
-	local mapsDisplay = {}
 	local maps = {}
-	local index = 1
+	for prefix, mapInput in Table.iter.pairsByPrefix(args, '') do
+		local mode = String.isNotEmpty(args[prefix .. 'mode']) and MapMode.get({args[prefix .. 'mode']}) or ''
 
-	while not String.isEmpty(args['map' .. index]) do
-		-- map game mode from mapXmode
-		local mapmode = ''
-		if not String.isEmpty(args['map' .. index .. 'mode']) then
-			mapmode = MapMode.get({args['map' .. index .. 'mode']})
-		end
+		mapInput = mw.text.split(mapInput, '|', true)
+		local display, link
 
-		-- map from mapX, might be pagename|displayname
-		local map = mw.text.split(args['map' .. index], '|', true)
-		-- maplink from mapXlink or first part of map or autolink
-		local maplink
-		if not String.isEmpty(args['map' .. index .. 'link']) then
-			maplink = args['map' .. index .. 'link']
-			map = map[1]
+		if String.isNotEmpty(args[prefix .. 'link']) then
+			link = args[prefix .. 'link']
+			display = mapInput[1]
 		else
-			maplink = map[1]
+			link = mapInput[1]
 			-- only check for a map page when map has only one part,
 			-- so no precise link is given
-			if map[2] == nil and Page.exists(maplink .. ' (map)') then
-				maplink = maplink .. ' (map)'
+			if mapInput[2] == nil and Page.exists(link .. ' (map)') then
+				link = link .. ' (map)'
 			end
-			map = map[2] or map[1]
+			display = mapInput[2] or mapInput[1]
 		end
+		link = mw.ext.TeamLiquidIntegration.resolve_redirect(link)
 
-		if index == 1 then
-			mapsDisplay = {Page.makeInternalLink({}, map .. mapmode, maplink)}
-		else
-			table.insert(mapsDisplay, '&nbsp;• ' ..
-				tostring(CustomLeague:_createNoWrappingSpan(
-					Page.makeInternalLink({}, map .. mapmode, maplink)
-				))
-			)
-		end
-		table.insert(maps, {['displayname'] = map, ['link'] = mw.ext.TeamLiquidIntegration.resolve_redirect(maplink)})
-		index = index + 1
+		table.insert(maps, {link = link, displayName = display, mode = mode})
 	end
 
-	return mapsDisplay, maps
+	_league.maps = maps
+
+	return maps
+end
+
+function CustomLeague:_displayMaps(maps)
+	local mapDisplay = function(map)
+		return tostring(CustomLeague:_createNoWrappingSpan(
+			Page.makeInternalLink({}, map.displayName .. map.mode, map.link)
+		))
+	end
+
+	return table.concat(
+		Table.mapValues(maps, function(map) return mapDisplay(map) end)
+		'&nbsp;• '
+	)
```

Comment:
This won't work, as the Center widget expects a table, not a string.
Could do something like
```suggestion
	local mapDisplay = function(index, map)
	    separator = index ~= 1 and '&nbsp;• ' or ''
		return index, separator .. tostring(CustomLeague:_createNoWrappingSpan(
			Page.makeInternalLink({}, map.displayName .. map.mode, map.link)
		))
	end

	return Table.map(maps, function(index, map) return mapDisplay(index, map) end)
```

Reply:
or just wrap it with {} in the return

Reply:
Does not work: https://liquipedia.net/ageofempires/User:SyntacticSalt/Sandbox/1

Reply:
will check when back home

Reply:
Full stacktrace when called from console:
```
-----Error in Widget:tryMake()-----
error = "Module:Infobox/League/Custom/dev:394: attempt to call a table value"
widget = table#1 {
  metatable = table#2
  ["children"] = table#3 {
    table#4 {
      metatable = table#5
      ["content"] = "Players",
    },
    table#6 {
      metatable = table#7
      ["content"] = table#8 {
      },
      ["name"] = "Number of Teams",
    },
    table#9 {
      metatable = table#7
      ["content"] = table#10 {
      },
      ["name"] = "Number of Players",
    },
    table#11 {
      metatable = table#5
      ["content"] = "Maps",
    },
  },
  ["context"] = table#12 {
    ["injector"] = table#13 {
      metatable = table#14
    },
  },
  ["id"] = "customcontent",
}
stack traceback:
	Module:Infobox/Widget:40: in function <Module:Infobox/Widget:36>
	[C]: ?
	Module:Infobox/League/Custom/dev:394: in function '_displayMaps'
	Module:Infobox/League/Custom/dev:98: in function <Module:Infobox/League/Custom/dev:52>
	(tail call): ?
	(tail call): ?
	[C]: in function 'xpcall'
	Module:Infobox/Widget:32: in function 'tryMake'
	Module:Infobox/Widget/Factory:23: in function 'work'
	Module:Infobox:63: in function 'build'
	Module:Infobox/League:234: in function <Module:Infobox/League:60>
	(tail call): ?
	console input:14: in function <console input:1>
	[C]: ?
```

Reply:
missing a `,`